io/pci/pci_info_lscfg.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2022 IBM
+# Author: Maram Srimannarayana Murthy<Maram.Srimannarayana.Murthy@ibm.com>
+
+"""
+PCI CFG Test.
+Needs to be run as root.
+"""
+
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import pci
+from avocado.utils import process
+
+
+class PciLscfgInfo(Test):
+    '''
+    Check lscfg info with lspci info
+    '''
+
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        smm = SoftwareManager()
+        for pkg in ["lsvpd", "pciutils"]:
+            if not smm.check_installed(pkg) and not smm.install(pkg):
+                self.cancel("{pkg} package is need to test")
+        if process.system("vpdupdate", ignore_status=True, shell=True):
+            self.fail("VPD Update fails")
+
+    def test(self):
+        '''
+        Capture data from lscfg and lspci then compare data
+        '''
+        error = []
+        for pci_addr in pci.get_pci_addresses():
+            self.log.info("Checking for PCI Address: %s\n\n", pci_addr)
+            pci_information_dict = pci.get_pci_info(pci_addr)
+            self.log.info(pci_information_dict)
+            cfg_output = pci.get_cfg(pci_addr)
+            self.log.info(cfg_output)
+            if cfg_output:
+
+                # Physical Slot Match
+                if(self.compare_phy_slot_lscfg_lspci(cfg_output['slot'],
+                   pci_information_dict['PhySlot'])):
+                    self.log.info("Physical Slot matched")
+                else:
+                    error.append("Physical slot info didn't match")
+
+                # Device ID match
+                if(self.compare_device_id_lspci_lscfg(
+                   cfg_output['subvendor_device'],
+                   pci_information_dict['Device'])):
+                    self.log.info("Device ID matched")
+                else:
+                    error.append("Device ID info didn't match")
+
+                # Subvendor ID Match
+                if(self.compare_sub_vendor_lspci_lscfg(
+                   cfg_output['subvendor_device'],
```

Comment:
@maramsmurthy 
how come cfg_output['subvendor_device'] can be used for both "Device_ID and Subvendor_ID"?
please check on it

Reply:
@Naresh-ibm 

[root@ltcden4-lp7 ~]# lscfg -vl 0135:90:00.0
  0135:90:00.0 fibre-channel U78CD.001.FZHAE88-P1-C5-T0
                                         fibre-channel (**10142281**)              
        Manufacturer Name.........IBM
        Machine Type-Model........ISP2812-based 64/32G Fibre Channel to PCIe
                                    Controller
        Engineering Change Level..P14641
        Field Replaceable Unit Number
        ..........................02CM909
        Manufacturer ID...........RFD2028L18792
        Part Number of assembly...02CM905
        Serial Number.............11S02CM905Y250NY07NEVJ
        Alterable ROM Level.......070101
        Customer Card ID Number.(CC)
        ..........................579C
        Device Specific.(YC)......2
        Location Code.(YL)........U78CD.001.FZHAE88-P1-C5-T0

Above Output has this value "10142281"
Device: 2281
SVendor:        1014
So captured these values respectively and compared with lspci output.